File: backend/models/casting.py
from sqlalchemy import ForeignKey, Table, Column
from sqlalchemy.sql.sqltypes import Integer, String, Float, Boolean, Date
from config.db import meta, engine

castings = Table("castings", meta, 
    Column("id", Integer, primary_key=True),
    Column("castingDate", Date),
    Column("name", String(255)),
    Column("castingDirector", Integer, ForeignKey("people.id")),
    Column("director", Integer, ForeignKey("people.id")),
    Column("inPerson", Boolean),
    Column("inProcess", Boolean),
    Column("notes", String(355)))

meta.create_all(engine)

File: frontend/lib/models/casting.dart
import 'package:frontend/models/person.dart';

class Casting {
  int id;
  DateTime castingDate;
  String name;
  Person castingDirector;
  Person director;
  bool inPerson;
  bool inProcess;
  String notes;

  Casting(
      this.id, this.castingDate, this.name, this.castingDirector, this.director, this.inPerson, this.inProcess, this.notes);
}
